apps/src/templates/policy_compliance/ParentalPermissionBanner/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment/moment';
+import {Provider} from 'react-redux';
+
+import {getStore} from '@cdo/apps/redux';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+import {EVENTS, PLATFORMS} from '@cdo/apps/lib/util/AnalyticsConstants';
+import color from '@cdo/apps/util/color';
+import i18n from '@cdo/locale';
+import getCurrentLocale from '@cdo/apps/util/currentLocale';
+import Notification from '@cdo/apps/templates/Notification';
+import {ParentalPermissionRequest} from '@cdo/apps/redux/parentalPermissionRequestReducer';
+import ParentalPermissionModal from '@cdo/apps/templates/policy_compliance/ParentalPermissionModal';
+
+interface ParentalPermissionBannerProps {
+  lockoutDate: string;
+}
+
+const ParentalPermissionBanner: React.FC<ParentalPermissionBannerProps> = ({
+  lockoutDate,
+}) => {
+  const currentUser = getStore().getState().currentUser;
+  const [consentStatus, setConsentStatus] = useState(
+    currentUser.childAccountComplianceState
+  );
+  const [showModal, setShowModal] = useState(false);
+  const formattedLockoutDate = moment(lockoutDate)
+    .lang(getCurrentLocale())
+    .format('ll');
+
+  const reportEvent = (eventName: string, payload: object = {}) => {
+    analyticsReporter.sendEvent(eventName, payload, PLATFORMS.STATSIG);
+  };
+
+  const handleModalShow = () => {
+    setShowModal(true);
+
+    reportEvent(EVENTS.CPA_PARENT_EMAIL_BANNER_CLICKED, {consentStatus});
+  };
+
+  const handleModalClose = (
+    parentalPermissionRequest: ParentalPermissionRequest | null | undefined
+  ) => {
+    setShowModal(false);
+
+    const newConsentStatus = parentalPermissionRequest
+      ? parentalPermissionRequest.consent_status
+      : consentStatus;
+
+    setConsentStatus(newConsentStatus);
+
+    reportEvent(EVENTS.CPA_PARENT_EMAIL_BANNER_CLOSED, {
+      consentStatus: newConsentStatus,
```

Comment:
Does the analytics library already include `in_section`?

Reply:
https://github.com/code-dot-org/code-dot-org/pull/58434/commits/eb58bb2047c42fafc9a9c50478276fb10785fc46